TykTechnologies/tyk-sync
# TT-5555 Adding OAS support

### Description
This PR adds OAS support. 

If an API has `is_oas` set to `true`, Tyk-sync will use `/api/apis/oas` endpoint to update the `id`. For the API creation, the normal `/api/apis` endpoint seems enough.


We fix the OAS ID modifying the `x-tyk-api-gateway.info.dbId` field with the new ID from API creation.


### How to test it
- Use tyk-sync dump command to dump OAS and tyk APIS
Example:
`./tyk-sync dump -d="http://localhost:3000" -s="7080fcd212a6489b64f6951f4b60e7b9" -t="examples/oas"`
It should dump all the APIs correctly.

- Manually delete all the APIs from the dashboard.
- Use tyk-sync sync command to re-create the APIs
Example:
`./tyk-sync sync -d="http://localhost:3000" -s="7080fcd212a6489b64f6951f4b60e7b9" -p="examples/oas" `
It should create the API's again.

- Use tky-sync sync command again. It should update the APIs instead of deleting and creating them again.


https://tyktech.atlassian.net/browse/TT-5555
